app/assets/javascripts/govuk_publishing_components/lib/single-consent-functions.js

```diff
@@ -0,0 +1,65 @@
+/* global GovSingleConsent */
+// = require govuk-single-consent/dist/singleconsent.iife.js
+
+(function (root) {
+  'use strict'
+  window.GOVUK = window.GOVUK || {}
+
+  window.GOVUK.singleConsent = {
+    init: function (callback) {
+      if (!this.useConsentApi()) {
+        return
+      }
+      callback = callback || this.apiCallBack
+      // determine where we are and set the consent api URL accordingly
+      if (!this.url) {
+        this.url = 'staging'
+        var environment = window.GOVUK.loadAnalytics.getEnvironment(window.GOVUK.analyticsGa4.core.trackFunctions.getHostname())
+        if (environment) {
+          this.url = environment.consentApiUrl
+        }
+      }
+      // create the consent API object
+      this.consentApiObj = new GovSingleConsent(callback, this.url)
+    },
+
+    useConsentApi: function () {
+      if (window.GOVUK.useSingleConsentApi) {
+        return true
+      }
+    },
+
+    apiCallBack: function (consents, consentsPreferencesSet, error) {
+      if (error) {
+        console.error('Single consent error: ', error, window.location)
+        return
+      }
+      if (consentsPreferencesSet) {
+        if (consents && consents.usage) {
+          window.GOVUK.triggerEvent(window, 'cookie-consent')
+        } else {
+          window.GOVUK.triggerEvent(window, 'hide-cookie-banner')
```

Comment:
To reduce the layout shift, is there a way that the 'Your preferences have been set / Hide this message' part of the banner HTML can just not be shown at all when the consent API exists? Currently there's a split second where you can see the message before it disappears.

I found it a bit jarring for it to appear for a split second then disappear, and when I first saw it I thought it was a bug with the code. I'm a bit worried that when user's see that, they will think that it's an important bit of information that's disappearing before they can read it.  It also feels like it could be flagged as an accessibility issue, though I haven't looked into this.

Reply:
Good shout. I think CLS is going to be an important part of this work. I'll investigate.

Reply:
I'm not sure I can reproduce the situation you're talking about, but I have identified a CLS problem when you've set preferences on another site and haven't been to GOV.UK before, which I think I've now fixed. Going to re-do the CLS testing again and add the numbers to the PR description.

Reply:
Numbers added. Seems to be no impact on CLS, which I think is mainly due to the recent no-JS changes to the banner 👍

Reply:
Also, what is the context behind no longer showing the "Cookies accepted" message? Or is this unintentional?

Reply:
Interesting - that shouldn't happen. I've just tried it by accepting cookies and it behaves as expected. Are you still able to reproduce?

Reply:
For what it's worth I can reproduce - haven't pulled the branch for a couple of days.

Reply:
https://github.com/alphagov/govuk_publishing_components/assets/31649453/f0015818-93ba-43d5-a93f-7a39854c6d2a

Reply:
@andysellick Yep I still have it on the latest version of the branch.

Reply:
@andysellick It's caused by:

`app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js`, line 20:

```JS
window.addEventListener('cookie-consent', this.$module.hideCookieMessage)
```

That code will hide the cookie banner when the `cookie-consent` event is triggered.

This `cookie-consent` event is then being triggered by `app/assets/javascripts/govuk_publishing_components/lib/single-consent-functions.js` line 33:

```JS
        if (consents && consents.usage) {
          window.GOVUK.triggerEvent(window, 'cookie-consent')
        } 
```

I think you can just remove the `cookie-consent` event listener that hides the cookie banner to fix it. What is it the context for having that listener?

Reply:
Nice, seems to work for me now 👍